### ISSUE TITLE
ResultQuery.fetchStream() fetches entire resultset at once

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11895]
Consider this code:

```java
Configuration c = create().configuration();

c.set(ExecuteListener.onRecordEnd(ctx -> System.out.println(ctx.record().get(TBook_ID()))));

try (Stream<B> s = c.dsl().fetchStream(TBook())) {
    Iterator<B> it = s.iterator();

    if (it.hasNext())
        it.next();
}
```

It fetches the entire result set into memory, printing:

```
1
2
3
4
```

When in fact, given that there's only one call to `it.next()`, it should fetch only one row. This seems to be a regression introduced with https://github.com/jOOQ/jOOQ/issues/4934 "Delay query execution until a Stream terminal op is called". The `fetchStream()` implementation does:

```java
    @Override
    default Stream<R> fetchStream() {
        return Stream.of(1).flatMap(i -> fetchLazy().stream());
    }
```

But flatmap just consumes the entire flatmapped stream at once

**Comment by lukaseder** [comment]
It seems this fixes the problem:

```java
    default Stream<R> fetchStream() {
        AtomicReference<Cursor<R>> r = new AtomicReference<>();
    
        // [#11895] Don't use the Stream.of(1).flatMap(i -> fetchLazy().stream())
        //          trick, because flatMap() will consume the entire result set
        return StreamSupport.stream(
            () -> {
                Cursor<R> c = fetchLazy();
                r.set(c);
                return c.spliterator();
            },
            Spliterator.IMMUTABLE | Spliterator.NONNULL | Spliterator.ORDERED,
            false
        ).onClose(() -> {
            JDBCUtils.safeClose(r.get());
        });
    }
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.10 (#11896)
